lib/rspec/bash/call_configuration.rb

```diff
@@ -2,36 +2,65 @@
 
 module Rspec
   module Bash
-    # Configuration of a stubbed command
     class CallConfiguration
       attr_reader :command
 
       def initialize(config_path, command)
         @config_path = config_path
-        @configuration = {}
+        @configuration = []
         @command = command
       end
 
       def set_exitcode(statuscode, args = [])
-        @configuration[args] ||= {}
-        @configuration[args][:statuscode] = statuscode
+        current_conf = create_or_get_conf(args)
+        current_conf[:statuscode] = statuscode
```

Comment:
Should we norm on whether we want to use exitcode or statuscode? Right now we're using both to mean the same thing throughout the library